### ISSUE TITLE
[no-inferrable-types] Boolean initialized to null tagged as inferrable

### ISSUE THREAD

**Issue by jp7837** [typescript-eslint/typescript-eslint#321]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-inferrable-types": 2
  }
}
```

```TS
let myBool: boolean = null;
```

**Expected Result**
No violations

**Actual Result**
myBool is tagged as inferrable

**Additional Info**
I know `strictNullTypes` should be used to make this more explicit, but if it's not enabled, should null-initialized booleans be flagged as inferrable?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.3.0` |
| `@typescript-eslint/parser`        | `1.3.0` |
| `TypeScript`                       | `3.3.3` |
| `ESLint`                           | `5.12.0` |
| `node`                             | `8.10.0` |

**Comment by bradzacher** [comment]
Reviewing the rule, its intention and documentation - the rule is intended to help reduce _unnecessary_ type annotations because the type is easily inferred from its literal value.

It does not validate any type annotations - it just checks:
- is the variable's initialisation value is simple enough? and
- does the init value's type match the type annotation?

In this case, the initialisation value (`null`), does not match the annotation (`boolean`), so the rule will not warn that the annotation is inferable, because it is not.

**Comment by jp7837** [comment]
@bradzacher sorry if the repro steps were confusing, I would also expect the rule would not warn, but the Actual Result is a `no-inferrable-types` violation.

**Comment by bradzacher** [comment]
oh, sorrry, I misunderstood your post!
Thanks for clarifying
